src/common/context/forms/index.tsx

```diff
@@ -64,47 +64,34 @@ export const FormsContextProvider: FunctionComponent = ({ children }) => {
   const setCurrentFormData = (data: any): void => {
     if (activeFormIndex === undefined)
       throw new Error("Trying to set form when there is no activeFormIndex");
-    const nextForms = [...forms];
-    const currentForm = forms[activeFormIndex];
-    nextForms.splice(activeFormIndex, 1, { ...currentForm, data });
-    setForms(nextForms);
+    setCurrentForm(data);
   };
 
   const setCurrentFormOwnership = ({ beneficiaryAddress, holderAddress }: Ownership): void => {
     if (activeFormIndex === undefined)
       throw new Error("Trying to set form when there is no activeFormIndex");
-    const nextForms = [...forms];
-    const currentForm = forms[activeFormIndex];
-    nextForms.splice(activeFormIndex, 1, {
-      ...currentForm,
-      ownership: { beneficiaryAddress, holderAddress },
-    });
-    setForms(nextForms);
+    setCurrentForm(undefined, { beneficiaryAddress, holderAddress });
   };
 
   const setCurrentFileName = (fileName: string): void => {
     if (activeFormIndex === undefined) return;
-    const updatedCurrentForm = {
-      ...currentForm,
-      fileName: fileName,
-    } as FormEntry;
-    const nextForms = [...forms];
-    nextForms.splice(activeFormIndex, 1, { ...updatedCurrentForm });
-    setForms(nextForms);
+    setCurrentForm(undefined, undefined, fileName);
```

Comment:
Can we change setCurrentForm to use destructured param instead of positional param? @rjchow will kill me if i let this though